synapse/storage/databases/main/stream.py

```diff
@@ -801,13 +801,66 @@ async def get_last_event_in_room_before_stream_ordering(
             before this stream ordering.
         """
 
-        last_row = await self.get_room_event_before_stream_ordering(
```

Comment:
Is `get_room_event_before_stream_ordering` just wrong? Should we be auditing its other call sites?